src/mps/abstractmps.jl

```diff
@@ -1734,9 +1733,13 @@ splitblocks(::typeof(linkinds), M::AbstractMPS; tol = 0) =
 #
 
 BroadcastStyle(MPST::Type{<:AbstractMPS}) = Style{MPST}()
+BroadcastStyle(::DefaultArrayStyle{1}, ::Style{MPST}) where {MPST<:AbstractMPS} = Style{MPST}()
+BroadcastStyle(::Style{MPST}, ::DefaultArrayStyle{1}) where {MPST<:AbstractMPS} = Style{MPST}()
 
 copyto!(ψ::AbstractMPS, b::Broadcasted) = copyto!(data(ψ), b)
 
+similar(::Broadcasted{Style{MPST}}, ::Type{ElType}, dims) where {N,ElType,MPST<:AbstractMPS} =
+    similar(Array{ElType}, dims)
```

Comment:
I tried to mimic the existing behaviour. Right now `psi .* xs` returns a `Vector{ITensor}`.

Reply:
Right, but that was just "by default" (it was just doing that automatically, so I would consider it undocumented and undefined), so this would be a good chance to think about what we want the behavior to be.

Reply:
It shouldn't really break peoples code to change the output type, since an MPS generally acts like a `Vector{ITensor}`. Hard to imagine someone was really relying on it being a `Vector{ITensor}` specifically.

Also we haven't registered `v0.2` yet since I still want to fix some performance regressions that were caused by the `IndexSet` deparametrization PR and also possibly add an ITensor slicing interface that I'm working on, so it's a good time for breaking changes.